File: app/src/main/java/coolpharaoh/tee/speicher/tea/timer/views/utils/display_amount_kind/DisplayAmountKindStrategy.java
package coolpharaoh.tee.speicher.tea.timer.views.utils.display_amount_kind;

import java.text.DecimalFormat;

public interface DisplayAmountKindStrategy {
    String getTextShowTea(double amount);

    int getImageResourceIdShowTea();

    String getTextCalculatorShowTea(float amountPerLiter, float liter);

    String getTextNewTea(double amount);

    static String getFormattedAmount(final double amount) {
        String text = "-";
        if (exist(amount)) {
            text = removeZerosFromAmount(amount);
        }
        return text;
    }

    static String removeZerosFromAmount(final double amount) {
        if (amount == (int) amount)
            return String.valueOf((int) amount);
        else {
            final DecimalFormat df = new DecimalFormat("#.#");
            return df.format(amount);
        }
    }

    static boolean exist(final double amount) {
        return amount != -500;
    }
}

File: app/src/main/java/coolpharaoh/tee/speicher/tea/timer/views/utils/display_amount_kind/DisplayAmountKindStrategy.kt
package coolpharaoh.tee.speicher.tea.timer.views.utils.display_amount_kind

import java.text.DecimalFormat

interface DisplayAmountKindStrategy {

    fun getTextShowTea(amount: Double): String

    fun getImageResourceIdShowTea(): Int

    fun getTextCalculatorShowTea(amountPerLiter: Float, liter: Float): String

    fun getTextNewTea(amount: Double): String

    companion object {
        @JvmStatic
        fun getFormattedAmount(amount: Double): String {
            var text = "-"
            if (exist(amount)) {
                text = removeZerosFromAmount(amount)
            }
            return text
        }

        fun removeZerosFromAmount(amount: Double): String {
            return if (amount == amount.toInt().toDouble()) amount.toInt().toString()
            else {
                val df = DecimalFormat("#.#")
                df.format(amount)
            }
        }

        fun exist(amount: Double): Boolean {
            return amount != -500.0
        }
    }
}